Adding JWT Access Token Authentication Option

jschroth opened the pull request:
Added option for Access Token and updated trino-go-client to v0.318.0

nineinchnick commented on pkg/trino/driver/driver.go:
```diff
@@ -57,10 +58,15 @@ func Open(settings models.TrinoDatasourceSettings) (*sql.DB, error) {
+	if settings.JWTAccessToken != "" {
+	    dsnlog = strings.Replace(dsnlog, settings.JWTAccessToken, "<scrubbedToken>", -1)
+	}
+	log.DefaultLogger.Info("Connecting to Trino", "dsn", dsnlog)
```
I think we should not log this at all.

jschroth replied:
I removed the logging altogether

nineinchnick commented on src/ConfigEditor.tsx:
```diff
@@ -35,6 +39,16 @@ export class ConfigEditor extends PureComponent<Props, State> {
             </InlineField>
           </div>
+          <div className="gf-form-inline">
+            <FormField
```
I don't know much about this UI part, so excuse my naive question - but why use FormField instead of InlineField?

jschroth replied:
I was following some other guides about the form development, but it seems like the recommendation is to use InlineField and then Input so I switched to those

nineinchnick commented on pkg/trino/models/settings.go:
```diff
@@ -14,6 +14,7 @@ type TrinoDatasourceSettings struct {
 	URL                 *url.URL           `json:"-"`
 	Opts                httpclient.Options `json:"-"`
 	EnableImpersonation bool               `json:"enableImpersonation"`
+	JWTAccessToken      string             `json:"jwtAccessToken"`
```
I'd name this just `accessToken` here any in other places.

jschroth replied:
Done

nineinchnick commented:
@cla-bot check

nineinchnick commented:
Also, please rebase, we updated the Trino Go driver in another PR that was just merged.

jschroth commented:
@nineinchnick thanks for the review. I think I addressed your comments. I also made the fields in UI a little more consistent.

 I was not sure if I should update the `CHANGELOG.md` and `package.json` as recommended in the `DEVELOPMENT.md` file, but please let me know if you want me to do those things as well.

nineinchnick commented:
@jschroth please rebase and I can merge this. You could update the `CHANGELOG.md` and `package.json` too, but I would handle this anyway.

jschroth commented:
Git is being very strange I feel like this whole PR is now messed up. I am going to try opening a new one from main with the hope that it fixes whatever the rebase issue is.

jschroth commented:
Moved to https://github.com/trinodb/grafana-trino/pull/274
